utils/utils.h

```diff
@@ -354,7 +354,11 @@ extern struct gbConf gbConf;
                      __FILE__, __FUNCTION__, __LINE__)
 
 # define  GB_STRCPY(dst, src, destbytes)                             \
-            gbStrcpy((dst), (src), (destbytes),                      \
+            gbStrcpy((dst), (src), (destbytes), strlen(src),         \
+                     __FILE__, __FUNCTION__, __LINE__)
+
+# define  GB_STRCAT(dst, src, destbytes)                             \
+            gbStrcat((dst), (src), (destbytes), strlen(src),         \
```

Comment:
> passing the strlen() of the source string as the destination
> length is pointless, and gcc-8 now warns about it:

@lxbsz but we are doing the same pointless thing right :-)

How about:

```
--- a/utils/utils.c                                 
+++ b/utils/utils.c                                 
@@ -477,7 +477,7 @@ gbStrcpy(char *dest, const char *src, size_t destbytes,                              
     if (n > (destbytes - 1))                       
       return NULL;                                 
                                                    
-    ret = strncpy(dest, src, n);                   
+    ret = memcpy(dest, src, n);                    
     /* strncpy NULL terminates if the last character is \0.  Therefore                                  
      * force the last byte to be \0                
      */                                            
@@ -485,3 +485,17 @@ gbStrcpy(char *dest, const char *src, size_t destbytes,                             
                                                    
     return ret;                                    
 }                                                  
+                                                   
+                                                   
+char *                                             
+gbStrcat(char *dest, const char *src, size_t destbytes,                                                 
+         const char *filename, const char *funcname, size_t linenr)                                     
+{                                                  
+    size_t n = strlen(src);                        
+                                                   
+    if (n > (destbytes - 1))                       
+      return NULL;                                 
+                                                   
+    /* strncat will add the NULL terminates */     
+    return memcpy(dest, src, n);                   
+} 
```

Reply:
Yeah, this looks good to me. Will update it.
Thanks.